crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -879,12 +861,15 @@ fn register_delete_link(
 
     // Just require that this link exists, don't need to check anything else about it here
     let mut validation_dependencies = validation_dependencies.lock();
-    validation_dependencies
+    let add_link_action = validation_dependencies
         .get(link_add_address)
         .and_then(|s| s.as_action())
         .ok_or_else(|| ValidationOutcome::DepMissingFromDht(link_add_address.clone().into()))?;
 
-    Ok(())
+    match add_link_action {
+        Action::CreateLink(_) => Ok(()),
+        _ => Err(ValidationOutcome::NotCreateLink(add_link_action.to_hash()).into()),
```

Comment:
I've added this check and interestingly the error code for it was already here... So I maybe this was here but got removed? I can understand baseless links wanting to be able to do creates with the base/target not being validated. But this isn't a user-supplied value, it's structural and it doesn't make sense for you to be able to DeleteLink an Update or something else right?

Also discussed here https://github.com/holochain/holochain/pull/3124#discussion_r1420427250